sources/ClangSharp.PInvokeGenerator/CSharp/CSharpOutputBuilder.VisitDecl.cs

```diff
@@ -268,6 +281,27 @@ public void BeginFunctionOrDelegate<TCustomAttrGeneratorData>(in FunctionOrDeleg
             }
         }
 
+        private void WriteSourceLocation(CXSourceLocation location, bool inline)
```

Comment:
I'm concerned this is going to end up very noisy across regenerations. The anonymous struct names/remappings is already a pain point for some users

Reply:
It's opt-in. It is useful for diagnostics (aka "where did this come from?"), but not so much for end-user of the bindings. I don't expect SharpGen-generated bindings or e.g. TerraFX.Interop to contain this info. But e.g. win32metadata and SharpGen might use this information to enrich logging or other forms of diagnostics.

Reply:
Right, I understand the benefit.

I was just iterating the concern around noise in practice. Many diffs would become nearly unreadable due to minor changes in information such as whitespace addition.

If it's being added here, a corresponding switch and basic test validating it works and doesn't regress should also be added.

Reply:
Done. I've added 2 tests that cover all important cases.